### PR TITLE
3214 - Files and links per Cycle: Show selected item for repository item

### PR DESCRIPTION
**Pull request by sorja**
https://github.com/openforis/fra-platform/assets/5508251/ac037fe2-c64b-431b-b26c-ea17e4c0784e

Next:
File upload progress indicator

### CODE REVIEW

**Comment by minotogna** [comment]
> repository.fileupload.mp4 
> 
> Next: File upload progress indicator

@sorja what happened to the add button for global items ?

**Comment by sorja** [comment]
> > repository.fileupload.mp4
> > Next: File upload progress indicator
> 
> @sorja what happened to the add button for global items ?

This video is recorded logged  in as National Correspondent

File: src/client/pages/CountryHome/Repository/EditForm/Actions/Actions.tsx [hunk 1]
```diff
@@ -34,27 +33,26 @@ const Actions: React.FC = () => {
...
+          label={t('editUser.done')}
           onClick={upsertRepositoryItem}
-          className={classNames('btn btn-primary', { disabled })}
-          type="button"
-        >
-          {/* TODO: Move to common.save */}
-          {t('editUser.done')}
-        </button>
+          disabled={disabledDone}
+        />
```
Comment (minotogna) [inline src/client/pages/CountryHome/Repository/EditForm/Actions/Actions.tsx:51]: @sorja can we please sort all props in all components of this pr? I will review again afterwards

File: src/client/store/ui/repository/actions/getFileMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { File } from 'meta/file'
+
+import { ThunkApiConfig } from 'client/store/types'
+
+type Props = CycleParams & {
+  fileUuid: string
+  uuid: string
+}
+
+export const getFileMetadata = createAsyncThunk<Omit<File, 'buffer'>, Props, ThunkApiConfig>(
```
Comment (minotogna) [inline src/client/store/ui/repository/actions/getFileMetadata.ts:15]: do we really need a new endpoint to get this FileMetadata, that I would call FileSummary ? those are not metadata, it's just a file without buffer. and add the type FileSummary in meta/file ?
Can this summary be fetched with the repository items directly ?

File: src/client/pages/CountryHome/Repository/EditForm/Actions/Actions.tsx (hunk 1 above)
Comment (sorja) [reply src/client/pages/CountryHome/Repository/EditForm/Actions/Actions.tsx:51]: Sure, added eslint rule to do this automatically in the future :)

**Review by minotogna — APPROVED** [review]

**Pull request merged**